src/validators/abstractions/base.validator.ts

```diff
@@ -0,0 +1,51 @@
+import { ObjectHasher } from "@blockr/blockr-crypto";
+import { DataAccessLayer } from "@blockr/blockr-data-access";
+import { logger } from "@blockr/blockr-logger";
+import { ValidationCondition } from "../concretes/validation.condition";
+import { IValidator } from "../interfaces/validator";
+
+export abstract class BaseValidator<IModel> implements IValidator<IModel> {
+    protected dataAccessLayer: DataAccessLayer;
+    protected objectHasher: ObjectHasher;
+    protected validationConditions: Array<ValidationCondition<IModel>> = [];
+
+    constructor(dataAccessLayer: DataAccessLayer, objectHasher: ObjectHasher) {
```

Comment:
Constructor can be protected.

```suggestion
    protected constructor(dataAccessLayer: DataAccessLayer, objectHasher: ObjectHasher) {
```